File: code_blocks/integrations/third-party-integrations/mparticle_7.dart
import 'package:mparticle_flutter/mparticle_flutter.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> login() async {
  final result = await MParticleFlutter.identity
      .login(IdentityRequest(userId: 'my_app_user_id'));
  final mPid = result.user.userId;
  await Purchases.collectDeviceIdentifiers();
  await Purchases.setMparticleID(mPid.toString());
}

File: code_blocks/integrations/third-party-integrations/mparticle_7.kt
import com.mparticle.MParticle
import com.mparticle.identity.IdentityApiRequest
import com.revenuecat.purchases.Purchases

val request = IdentityApiRequest.withEmptyUser().also {
    it.userId = "my_app_user_id"
}

MParticle.getInstance()?.identity?.login(request)?.addSuccessListener { result ->
    val mPid = result.user.id
    Purchases.sharedInstance.attribution.collectDeviceIdentifiers()
    Purchases.sharedInstance.attribution.setMparticleID(mPid.toString())
}

